### PR TITLE
SOC-5361 [social-rdbms] My connection page is too slow

### PR DESCRIPTION
**Pull request by boubaker**
- Move Common Transformation operations (Entity to DTO) into Utils class to simplify transformations in all DAO and avoid getting DTO from top layer services
- Transform List to Set in order ot be able to fetch multiple collections at the same time using JPQL
- Fix the identities count in AllPeople page and modify query to retrieve identities with relationship at the same time in same DTO
- Modify Connections queries to use NamedQueries instead of building query when possible (with minimal modifications).

### CODE REVIEW

**Review by thomasdelhomenie — CHANGES_REQUESTED** [review]

File: lib/src/main/java/org/exoplatform/social/addons/storage/entity/IdentityEntity.java [hunk 1]
```diff
@@ -56,6 +54,24 @@
 @Table(name = "SOC_IDENTITIES")
 @NamedQueries({
         @NamedQuery(
+                name = "SocIdentity.findAllUserIdentities",
+                query = "SELECT distinct identity " +
+                " FROM SocIdentityEntity identity "  +
+                " WHERE   identity.deleted = FALSE " +
+                "     AND identity.enabled = TRUE " +
+                "     AND identity.id != :identityId " +
```
Comment (boubaker) [inline lib/src/main/java/org/exoplatform/social/addons/storage/entity/IdentityEntity.java]: Done, thanks.

File: lib/src/main/java/org/exoplatform/social/addons/storage/entity/IdentityEntity.java [hunk 2]
```diff
@@ -56,6 +54,24 @@
...
+                "     AND identity.providerId= :providerId "
```
Comment (boubaker) [inline lib/src/main/java/org/exoplatform/social/addons/storage/entity/IdentityEntity.java]: Done, thanks.

File: lib/src/main/java/org/exoplatform/social/addons/storage/Utils.java [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+import org.exoplatform.social.core.identity.model.Identity;
+import org.exoplatform.social.core.identity.model.IdentityWithRelationship;
+import org.exoplatform.social.core.identity.model.Profile;
+import org.exoplatform.social.core.identity.provider.OrganizationIdentityProvider;
+import org.exoplatform.social.core.identity.provider.SpaceIdentityProvider;
+import org.exoplatform.social.core.model.AvatarAttachment;
+import org.exoplatform.social.core.relationship.model.Relationship;
+import org.exoplatform.social.core.service.LinkProvider;
+
+public class Utils {
```
Comment (boubaker) [inline lib/src/main/java/org/exoplatform/social/addons/storage/Utils.java]: Done, thanks. (EntityConverterUtils, Utils is used to know that it's an utility class with static methods)

File: lib/src/main/java/org/exoplatform/social/addons/storage/RDBMSRelationshipStorageImpl.java [hunk 4]
```diff
@@ -397,6 +391,31 @@ public int getOutgoingCountByFilter(Identity existingIdentity, ProfileFilter pro
     }
   }
 
+  private List<Identity> searchConnections(Identity owner, Type status, long offset, long limit, Sorting sorting) {
+    // FIXME: Sorting is not applied,
```
Comment (boubaker) [inline lib/src/main/java/org/exoplatform/social/addons/storage/RDBMSRelationshipStorageImpl.java]: Done, thanks.

**Review by thomasdelhomenie — APPROVED** [review]

**Comment by boubaker** [comment]
Merged

**Pull request closed without merging**